### PR TITLE
Fixed --registry-server with --image bug.

### PR DESCRIPTION
**Pull request by haroonf**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/containerapp/azext_containerapp/_up_utils.py [hunk 1]
```diff
@@ -278,12 +278,11 @@ def create(self, no_registry=False):
 
     def create_acr_if_needed(self):
         if self.should_create_acr:
-             logger.warning(
-                 f"Creating Azure Container Registry {self.acr.name} in resource group "
-                 f"{self.acr.resource_group.name}"
-             )
-             self.create_acr()
-
+            logger.warning(
```
Comment (panchagnula) [inline src/containerapp/azext_containerapp/_up_utils.py:281]: any reason the notification about creation is after the create operation completes?
what would be the experience for the user if this operation fails - they won't see any "Creating.." notification & just the error message? That seems wrong.

File: src/containerapp/azext_containerapp/_up_utils.py [hunk 2]
```diff
@@ -386,12 +385,12 @@ def _get_dockerfile_content_from_repo(
...
+    except Exception as e:
+        raise ValidationError(f"Could not find repo {repo_url}") from e
     try:
         files = r.get_contents(context_path, ref=branch)
-    except:
-        raise ValidationError(f"Could not find branch {branch}")
+    except Exception as e:
+        raise ValidationError(f"Could not find branch {branch}") from e
     for f in files:
         if f.path == dockerfile or f.path.endswith(f"/{dockerfile}"):
```
Comment (panchagnula) [inline src/containerapp/azext_containerapp/_up_utils.py:395]: is this path (/) work for all OS types? as in if this is local path , might change based on OS?

File: src/containerapp/azext_containerapp/_up_utils.py (hunk 1 above)
Comment (haroonf) [reply src/containerapp/azext_containerapp/_up_utils.py:281]: The self.create_acr is the function that creates the acr, the message happens right before the operation begins

File: src/containerapp/azext_containerapp/_up_utils.py (hunk 2 above)
Comment (haroonf) [reply src/containerapp/azext_containerapp/_up_utils.py:395]: I believe this is from github, so shouldn't change anything. @StrawnSC let me know if I am wrong

Comment (StrawnSC) [reply src/containerapp/azext_containerapp/_up_utils.py:395]: @panchagnula these are paths from the GitHub API they aren't OS-dependent

**Review by StrawnSC — APPROVED** [review]
Thanks for the fix here, Rune

**Comment by StrawnSC** [comment]
Approving this since it prevents the weird exception when using --registry-url and --image (without ACR creds). Although we may want to consider just not allowing the user to provide a --registry-url with --image (w/o source or repo) if  we require that the image be a full URL with the registry in it

**Pull request merged**